src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb.c

```diff
@@ -1602,15 +1658,27 @@ static void lb_on_response_received_locked(grpc_exec_ctx *exec_ctx, void *arg,
             }
             grpc_grpclb_destroy_serverlist(serverlist);
           } else { /* new serverlist */
+            /* record the old status */
+            /* TODO(juanlishen): this assumes that num_servers never decreases
+             * to 0 once it's positive */
+            bool first_serverlist = glb_policy->serverlist == NULL;
```

Comment:
Instead of declaring this boolean, I suggest moving the code from line 1674 into an `else` block on line 1668.